File: offer/python3/offer_0004.py
from typing import List

class Solution:
    def findNumberIn2DArray(self, matrix: List[List[int]], target: int) -> bool:
        if len(matrix) == 0:
            return False
        i, j = 0, len(matrix[0]) - 1
        while i < len(matrix) and j >= 0:
            if matrix[i][j] > target:
                j -= 1
            elif matrix[i][j] < target:
                i += 1
            else:
                return True;
        return False;

File: offer/rust/offer_0004.rs
impl Solution {
    pub fn find_number_in2_d_array(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        if matrix.len() == 0 {
            return false
        }
        let mut i = 0;
        let mut j = matrix[0].len() as i32 - 1;
        while i < matrix.len() && j >= 0 {
            if matrix[i][j as usize] > target {
                j -= 1;
            } else if matrix[i][j as usize] < target {
                i += 1;
            } else {
                return true;
            }
        }
        false
    }
}